### PR TITLE
Dialect: add scala3x, move type placeholder there

### PR DESCRIPTION
**Pull request by kitbellew**
Unblocks #2884.

### CODE REVIEW

File: scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala [hunk 1]
```diff
@@ -789,10 +789,11 @@ object Dialect extends InternalDialect {
   }
   private lazy val standardPairs = Seq[sourcecode.Text[Dialect]](
     Dotty,
+    Scala3x,
```
Comment (tgodzik) [inline scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala:792]: Scala3Future ?

Comment (kitbellew) [reply scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala:792]: sorry, pulled the trigger too early. let me know if you'd like me to rename; tried to make it shorter (plus, some repos -- such as intellij -- use `x` to denote a future version).

File: tests/jvm/src/test/scala-2.13/scala/meta/tests/api/SurfaceSuite.scala [hunk 2]
```diff
@@ -105,6 +105,7 @@ class SurfaceSuite extends FunSuite {
       |scala.meta.dialects.Scala30 *
       |scala.meta.dialects.Scala31 *
       |scala.meta.dialects.Scala32 *
+      |scala.meta.dialects.Scala3x *
```
Comment (kitbellew) [inline tests/jvm/src/test/scala-2.13/scala/meta/tests/api/SurfaceSuite.scala:108]: @tgodzik future here

File: tests/shared/src/test/scala/scala/meta/tests/prettyprinters/PublicSuite.scala [hunk 3]
```diff
@@ -94,25 +94,29 @@ class PublicSuite extends TreeSuiteBase {
...
+    assertNoDiff(scala.meta.dialects.Scala31.toString, "Scala32")
   }
 
   test("scala.meta.dialects.Scala32.toString") {
-    assertNoDiff(scala.meta.dialects.Scala32.toString, "Scala3")
+    assertNoDiff(scala.meta.dialects.Scala32.toString, "Scala32")
   }
 
-  test("scala.meta.dialects.Dotty.toString") {
+  test("scala.meta.dialects.Scala3x.toString") {
```
Comment (kitbellew) [inline tests/shared/src/test/scala/scala/meta/tests/prettyprinters/PublicSuite.scala:112]: future here as well (in the test name!)

**Review by kitbellew — COMMENTED** [review]
@tgodzik here the two places which are failing tests for your x->Future change.

**Pull request merged**